### PR TITLE
OWLS-84562 - added tests for Namespace management enhancements

### PR DESCRIPTION
**Pull request by marinakog**
Jenkins : https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/2210/

Added tests to cover [https://jira.oraclecorp.com/jira/browse/OWLS-82046]  (https://github.com/oracle/weblogic-kubernetes-operator/pull/1860)
the tests don't cover Dedicated method, it is covered by other testclass
Testcases:
1. Create operator managing domain namespaces by SelectLabels
a. create namespace with specified label, make sure domain is started and operator can scale it
b. verify that domain with namespace with different label is not started/managed by operator, verify that as soon as namespace will be assigned the label, managed by operator, domain will be started and managed by operator.
c. verify that SelectLabel override domainNamespaces option from values and domain is not started even domainNamespace option is specified. 
d. verify that domain in default namespace is not started 
e. verify that operator can add/remove namespaces by upgrading helm chart and switch to different select method (RegExp)
f. verify that another operator (using List for domainNamespaces selection ) failed to share same namespace

2. Create operator managing domain namespaces by Expression searching for namespaces name starting with xoxoxo
a. create two namespace with namespaces staring with xoxoxo, make sure domains are started and operator can scale it
b. verify that domain with namespace with different name is not started/managed by operator
c. verify that RegExp override domainNamespaces option from values and domain is not started even domainNamespace option is specified. 
d. verify that domain in default namespace is not started 
e. verify that operator can add/remove namespaces by upgrading helm chart and switch to different select method (SelectLabels)
f. verify that another operator (using List for domainNamespaces selection ) failed to share same namespace

3. Create namespace before operator with setting enableRoleClusterBinding=false, and SelectLabel using that namespace, verify that domain can't be started , upgrade operator with reuse values, verify domain is started after upgrade.

### CODE REVIEW

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItManageNs.java [hunk 1]
```diff
@@ -0,0 +1,824 @@
...
+    }
+  }
+
+  /**
+   * Install WebLogic operator and wait up to two minutes until the operator pod is ready.
+   *
+   * @param operNamespace the operator namespace in which the operator will be installed
+   * @param opReleaseName the operator release name
+   * @param enableClusterRoleBinding operator cluster role binding
+   * @param domainNamespace the list of the domain namespaces which will be managed by the operator
```
Comment (sankarpn) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItManageNs.java:689]:
```suggestion
   * @param domainNamespace the list of the domain namespaces managed by the operator
```

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItManageNs.java [hunk 2]
```diff
@@ -0,0 +1,824 @@
...
+   * @return the operator Helm installation parameters
+   */
+  private static HelmParams installOperatorHelmChart(String opReleaseName, String operNamespace,
+                                                     String domainNsSelectionStrategy,
+                                                     String domainNsSelector,
+                                                     boolean enableClusterRoleBinding,
+                                                     String... domainNamespace) {
+    LoggingFacade logger = getLogger();
```
Comment (sankarpn) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItManageNs.java:697]: You can define a class level logger and use it in all the methods

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Kubernetes.java [hunk 3]
```diff
@@ -703,6 +704,34 @@ public static boolean createNamespace(String name) throws ApiException {
...
   }
 
+  // --------------------------- namespaces -----------------------------------
+  /**
+   * Create a Kubernetes namespace.
+   *
+   * @param name the name of the namespace
+   * @param labels list of labels for the namespace
+   * @return true on success, false otherwise
+   * @throws ApiException if Kubernetes client API call
```
Comment (sankarpn) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Kubernetes.java:714]: incomplete sentence

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Kubernetes.java [hunk 4]
```diff
@@ -760,6 +789,15 @@ public static void replaceNamespace(V1Namespace ns) throws ApiException {
    * @throws ApiException if Kubernetes client API call fails
    */
   public static List<String> listNamespaces() throws ApiException {
+    return listNamespaces(null);
+  }
+
+  /**
+   * List namespaces in the Kubernetes cluster.
```
Comment (sankarpn) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Kubernetes.java:796]:
```suggestion
   * List namespaces in the Kubernetes cluster matching the label selector.
```

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Kubernetes.java [hunk 5]
```diff
@@ -813,6 +851,39 @@ public static V1NamespaceList listNamespacesAsObjects() throws ApiException {
     return namespaceList;
   }
 
+  /**
+   * return namespace by provided name in the Kubernetes cluster as V1Namespace object.
```
Comment (sankarpn) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Kubernetes.java:855]:
```suggestion
   * Return namespace object for the given name from the Kubernetes cluster as V1Namespace object.
```

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Kubernetes.java [hunk 6]
```diff
@@ -813,6 +851,39 @@ public static V1NamespaceList listNamespacesAsObjects() throws ApiException {
...
+   * @name name of namespace.
+   * @return V1Namespace of Namespace in the Kubernetes cluster
```
Comment (sankarpn) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Kubernetes.java:857]:
```suggestion
   * @return V1Namespace  Namespace object from the Kubernetes cluster
```

**Review by sankarpn — CHANGES_REQUESTED** [review]

**Review by vanajamukkara — CHANGES_REQUESTED** [review]
still reviewing...

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItManageNs.java [hunk 7]
```diff
@@ -0,0 +1,824 @@
...
+/**
+ * Simple JUnit test file used for testing operator namespace management,
+ * Dedicated usecase is covered by other test class.
+ */
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+@DisplayName("Test operator namespace management usability using Helm chart")
+@IntegrationTest
+class ItManageNs {
+
+  private static String opNamespace = null;
```
[the remark this answers is not in the document]
Comment (marinakog) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItManageNs.java]: fixed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItManageNs.java [hunk 8]
```diff
@@ -0,0 +1,824 @@
...
+
+    // get a unique operator 3 namespace
+    logger.info("Getting a unique namespace for operator 3");
+    assertNotNull(namespaces.get(6), "Namespace list is null");
+    op3Namespace = namespaces.get(6);
+
+    // get a unique operator 4 namespace
+    logger.info("Getting a unique namespace for operator 4");
+    assertNotNull(namespaces.get(7), "Namespace list is null");
+    op4Namespace = namespaces.get(7);
```
[the remark this answers is not in the document]
Comment (marinakog) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItManageNs.java]: fixed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItManageNs.java [hunk 9]
```diff
@@ -0,0 +1,824 @@
...
+
+    // docker login and push image to docker registry if necessary
+    dockerLoginAndPushImageToRegistry(miiImage);
```
[the remark this answers is not in the document]
Comment (marinakog) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItManageNs.java]: removed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItManageNs.java (hunk 2 above)
Comment (marinakog) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItManageNs.java]: done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Kubernetes.java (hunk 3 above)
Comment (marinakog) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Kubernetes.java]: fixed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Kubernetes.java (hunk 4 above)
Comment (marinakog) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Kubernetes.java]: fixed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Kubernetes.java (hunk 5 above)
Comment (marinakog) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Kubernetes.java]: fixed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Kubernetes.java (hunk 6 above)
Comment (marinakog) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Kubernetes.java]: fixed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItManageNs.java [hunk 10]
```diff
@@ -0,0 +1,824 @@
...
+          op3Namespace, "List",
+          null, true, domainNamespace);
+      assertNull(opHelmParams3, "Operator helm chart sharing same NS with other operator did not fail");
+    } catch (org.opentest4j.AssertionFailedError ex) {
+      //expecting to fail
+      logger.info("Helm installation failed as expected " + ex.getMessage());
+    }
+  }
+
+  private void checkDomainNotStartedInDefaultNS(String domainNsSelectionStrategy) {
```
[the remark this answers is not in the document]
Comment (marinakog) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItManageNs.java]: fixed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItManageNs.java [hunk 11]
```diff
@@ -0,0 +1,824 @@
...
+      + " using expression namespace management")
+  public void testNsManageByExp() {
+    //create domain namespace
+    String manageByExp1NS = "test-" +  domain1Namespace;
+    String manageByExp2NS = "test-" +  domain2Namespace;
+    String manageByExpDomain1Uid = "test-" + domain1Uid;
+    String manageByExpDomain2Uid = "test-" + domain2Uid;
+    String manageByLabelNS = "weblogic1" + domain1Namespace;
+    String manageByLabelDomainUid = "weblogic1" + domain1Uid;
+    String domain3NS = "atest-" +  domain1Namespace;
```
[the remark this answers is not in the document]
Comment (marinakog) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItManageNs.java]: fixed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItManageNs.java [hunk 12]
```diff
@@ -0,0 +1,824 @@
...
+
+    logger.info("Installing and verifying domain2");
+    createSecrets(manageByExp2NS);
+    assertTrue(startDomain(manageByExp2NS, manageByExpDomain2Uid),
+        "operator can start or verify domain in namespace " + manageByExp2NS);
+    checkOperatorCanScaleDomain(op2Namespace, manageByExpDomain2Uid);
+  }
+
+
+  private boolean startDomain(String domainNamespace, String domainUid) {
```
[the remark this answers is not in the document]
Comment (marinakog) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItManageNs.java]: fixed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItManageNs.java [hunk 13]
```diff
@@ -0,0 +1,824 @@
...
+
+    // create and verify the domain
+    logger.info("Creating and verifying model in image domain");
+    try {
+      Domain domain = createDomainCRD(domainNamespace, domainUid);
+      createVerifyDomain(domainNamespace, domainUid, miiImage, domain);
+      return true;
+    } catch (Exception ex) {
```
[the remark this answers is not in the document]
Comment (marinakog) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItManageNs.java]: fixed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItManageNs.java [hunk 14]
```diff
@@ -0,0 +1,824 @@
...
+
+    //verify that domainNamespaces field will be ignored and domain will not start for specific NS and default
+    checkPodNotCreated(domain3Uid + adminServerPrefix, domain3Uid, domain3Namespace);
+    checkDomainNotStartedInDefaultNS("RegExp");
+    //verify that operator can't start domain if namespace does not start from test
+    createSecrets(domain3NS);
+    checkPodNotCreated(domain3Uid + adminServerPrefix, domain3Uid, domain3NS);
+
+    // install  operator sharing same domain
+    checkSecondOperatorFailedToShareSameNS(manageByExp1NS);
```
[the remark this answers is not in the document]
Comment (marinakog) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItManageNs.java:212]: fixed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItManageNs.java [hunk 15]
```diff
@@ -0,0 +1,824 @@
...
+        .domainNamespaceLabelSelector("mytest")
+        .domainNamespaceSelectionStrategy("LabelSelector");
+
+    assertTrue(upgradeAndVerifyOperator(op2Namespace, opParams));
+
+    //verify domain is started
+    createSecrets(manageByLabelNS);
+    assertTrue(startDomain(manageByLabelNS,manageByLabelDomainUid));
+    checkOperatorCanScaleDomain(op2Namespace,manageByLabelDomainUid);
+    //check operator can't manage anymore manageByExpNS
```
[the remark this answers is not in the document]
Comment (marinakog) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItManageNs.java]: fixed

**Comment by marinakog** [comment]
newest results
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/2263

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Kubernetes.java [hunk 16]
```diff
@@ -813,6 +851,39 @@ public static V1NamespaceList listNamespacesAsObjects() throws ApiException {
     return namespaceList;
   }
 
+  /**
+   * Return Namespace object for the given name from the Kubernetes cluster as V1Namespace object..
```
Comment (sankarpn) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Kubernetes.java:855]:
```suggestion
   * Return Namespace object for the given name from the Kubernetes cluster as V1Namespace object.
```

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItManageNs.java [hunk 17]
```diff
@@ -0,0 +1,824 @@
...
+  /**
+   * Install WebLogic operator and wait up to two minutes until the operator pod is ready.
+   *
+   * @param operNamespace the operator namespace in which the operator will be installed
+   * @param opReleaseName the operator release name
+   * @param enableClusterRoleBinding operator cluster role binding
+   * @param domainNamespace the list of the domain namespaces which will be managed by the operator
+   * @return the operator Helm installation parameters
+   */
+  private static HelmParams installOperatorHelmChart(String opReleaseName, String operNamespace,
```
[the remark this answers is not in the document]
Comment (marinakog) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItManageNs.java]: fixed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Kubernetes.java (hunk 16 above)
Comment (marinakog) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Kubernetes.java]: fixed

**Review by sankarpn — APPROVED** [review]
LGTM

**Review by vanajamukkara — CHANGES_REQUESTED** [review]
left few comments, but please modify the code based on what we discussed..

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItManageNs.java [hunk 18]
```diff
@@ -0,0 +1,633 @@
...
+    // install and verify operator set to manage domains based on LabelSelector strategy,
+    // domainNamespaces set to domain4 will be ignored
+    opHelmParams[0] = installAndVerifyOperator(OPERATOR_RELEASE_NAME,
+        opNamespaces[0], "LabelSelector",
+        OPERATOR_RELEASE_NAME, true, manageByLabelDomain1NS);
+
+    logger.info("Installing and verifying domain1");
+    createSecrets(manageByLabelDomain1NS);
+    assertTrue(createDomainCrdAndVerifyDomainIsRunning(manageByLabelDomain1NS, manageByLabelDomain1Uid),
+        "can't start or verify domain in namespace " + manageByLabelDomain1NS);
```
[the remark this answers is not in the document]
Comment (marinakog) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItManageNs.java]: done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItManageNs.java [hunk 19]
```diff
@@ -0,0 +1,633 @@
...
+  @Order(2)
+  @DisplayName("install operator helm chart and domain, "
+      + " using label namespace management")
+  public void testNsManageByLabel() {
+    String manageByLabelDomain1NS = domainNamespaces[0];
+    String manageByLabelDomain2NS = domainNamespaces[1];
+    String manageByExpDomainUid = "weblogic2" + domainNamespaces[1];
+    String manageByExpDomainNS = "weblogic2" + domainNamespaces[1];
+    String manageByLabelDomain1Uid = domainsUid[0];
+    String manageByLabelDomain2Uid = domainsUid[1];
```
[the remark this answers is not in the document]
Comment (marinakog) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItManageNs.java]: done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItManageNs.java [hunk 20]
```diff
@@ -0,0 +1,633 @@
...
+        .domainNamespaceRegExp("^" + "weblogic2");
+
+    assertTrue(upgradeAndVerifyOperator(opNamespaces[0], opParams));
+
+    //verify domain is started in namespace with name starting with weblogic* and operator can scale it.
+    createSecrets(manageByExpNS);
+    assertTrue(createDomainCrdAndVerifyDomainIsRunning(manageByExpNS,manageByExpDomainUid));
+    checkOperatorCanScaleDomain(opNamespaces[0],manageByExpDomainUid);
+    //verify operator can't manage anymore domain running in the namespace with label
+    assertTrue(isOperatorFailedToScaleDomain(opNamespaces[0], manageByLabelDomainUid, manageByLabelNS),
```
[the remark this answers is not in the document]
Comment (marinakog) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItManageNs.java]: done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/CommonTestUtils.java [hunk 21]
```diff
@@ -453,6 +454,133 @@ public static HelmParams installAndVerifyOperator(String opNamespace,
     return opHelmParams;
   }
 
+  /**
```
[the remark this answers is not in the document]
Comment (marinakog) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/CommonTestUtils.java:530]: done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/CommonTestUtils.java [hunk 22]
```diff
@@ -453,6 +454,133 @@ public static HelmParams installAndVerifyOperator(String opNamespace,
...
+    logger.info("Created service account: {0}", opReleaseName + "-sa");
+
+
+    // get operator image name
+    String operatorImage = getOperatorImageName();
+    assertFalse(operatorImage.isEmpty(), "operator image name can not be empty");
+    logger.info("operator image name {0}", operatorImage);
+
+
+    V1SecretList listSecrets = oracle.weblogic.kubernetes.actions.TestActions.listSecrets(operNamespace);
```
[the remark this answers is not in the document]
Comment (marinakog) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/CommonTestUtils.java]: done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItManageNs.java [hunk 23]
```diff
@@ -0,0 +1,824 @@
...
+                operNamespace,
+                condition.getElapsedTimeInMS(),
+                condition.getRemainingTimeInMS()))
+        .until(assertDoesNotThrow(() -> operatorRestServiceRunning(operNamespace),
+            "operator external service is not running"));
+
+    return opHelmParams;
+  }
+
+  private void checkPodNotCreated(String podName, String domainUid, String domNamespace) {
```
[the remark this answers is not in the document]
Comment (marinakog) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItManageNs.java:608]: done

**Comment by marinakog** [comment]
latest run https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/2282/

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItManageNs.java [hunk 24]
```diff
@@ -0,0 +1,614 @@
...
+        .externalRestEnabled(true)
+        .externalRestHttpsPort(externalRestHttpsPort)
+        .domainNamespaceSelectionStrategy("RegExp")
+        .domainNamespaceRegExp("^" + "weblogic");
+
+    assertFalse(upgradeAndVerifyOperator(opNamespaces[0], opParams), "Upgrade does not fail when adding domain,"
+        + " managed by other operator");
+  }
+
+  private void deleteDomainCrd(String domainNS, String domainUid) {
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItManageNs.java:384]: Here we are not deleting Domain CRD which is a global resource created by Operator. Actually we are deleting Domain Resource ( domain ).  Modify the method name to  deleteDomainResource()

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItManageNs.java [hunk 25]
```diff
@@ -0,0 +1,614 @@
...
+    logger.info("Creating and verifying model in image domain");
+    Domain domain = createDomainCRD(domainNamespace, domainUid);
+    assertDoesNotThrow(() -> createVerifyDomain(domainNamespace, domainUid, miiImage, domain));
+    return true;
+  }
+
+  /**
+   * Create a model in image domain crd.
+   */
+  private Domain createDomainCRD(String domainNamespace, String domainUid) {
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItManageNs.java:507]: Again here we are creating Domain Resource not CRD. Modify the method name to createDomainResource()

**Review by anpanigr — COMMENTED** [review]
Minor review comments.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItManageNs.java [hunk 26]
```diff
@@ -0,0 +1,614 @@
...
+   * Try to start another domain with namespace manageByExp3NS. Verify it is not started by operator
+   * and value of domainNamespace is ignored for both specific namespace and default.
+   * Add another operator using domainNamespaces sharing same namespace and verify it fails to install.
+   * Upgrade helm chart to switch to using LabelSelector, start domain and verify operator can scale it.
+   */
+  @Test
+  @Order(1)
+  @DisplayName("install operator helm chart and domain, "
+      + " using expression namespace management")
+  public void testNsManageByExp() {
```
[the remark this answers is not in the document]
Comment (marinakog) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItManageNs.java]: done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItManageNs.java [hunk 27]
```diff
@@ -0,0 +1,614 @@
...
+   * Call upgrade operator with reuse values to enable management for ns1
+   * Deploy a custom domain resource in the namespace ns1 with label1
+   * and verify all server pods in the domain were created and ready.
+   * Verify operator is able to manage this domain by scaling.
+   */
+  @Test
+  @Order(3)
+  @DisplayName("install operator helm chart and domain, "
+      + " with enableClusterRoleBinding")
+  public void testSwitchRbac() {
```
[the remark this answers is not in the document]
Comment (marinakog) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItManageNs.java]: done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItManageNs.java [hunk 28]
```diff
@@ -0,0 +1,614 @@
...
+   * Add another operator using domainNamespaces sharing same namespace and verify it fails to install.
+   * Upgrade operator to replace namespace management using RegExp namespaces.
+   * Verify it can manage added domain and can't manage old domain by scaling .
+   * Verify that upgrade helm fail if try to add domain, managed by other operator.
+   */
+  @Test
+  @Order(2)
+  @DisplayName("install operator helm chart and domain, "
+      + " using label namespace management")
+  public void testNsManageByLabel() {
```
[the remark this answers is not in the document]
Comment (marinakog) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItManageNs.java]: done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItManageNs.java (hunk 25 above)
Comment (marinakog) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItManageNs.java]: done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItManageNs.java (hunk 24 above)
Comment (marinakog) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItManageNs.java]: done

**Review by anpanigr — APPROVED** [review]
lgtm

**Review by vanajamukkara — APPROVED** [review]
LG

**Pull request merged**